File: codewar/V1/Day-438/perfectSquare.js
/**
 *
 * @param {Number} sq
 * @returns Number ???
 */
function findNextSquare(sq) {
  const isSquare = (n) => {
    return n > 0 && Math.sqrt(n) % 1 === 0;
  };

  if (isSquare(sq)) {
    return Math.pow(Math.sqrt(sq) + 1, 2);
  }

  return -1;
}

// console.log(findNextSquare(121));

/* --------------------------------- oneline -------------------------------- */

/**
 *
 * @param {Number} sq
 */
function findNextSquareOneLine(sq) {
  return Math.sqrt(sq) % 1 !== 0 ? -1 : Math.pow(Math.sqrt(sq) + 1, 2);
}

console.log(findNextSquareOneLine(121));

const test =
  Math.pow(2, 2) === 4 ? console.log("helllo daniel") : console.log("a");

File: codewar/V1/Day-438/perfect_square.py
import math


def is_square(n):
    return n > 0 and math.sqrt(n) % 1 == 0


def find_next_square(sq):
    if is_square(sq):
        return math.pow(math.sqrt(sq) + 1, 2)
    return -1


# print(find_next_square(122))

# -------------------------------- better one -------------------------------- #


def second(sq):
    return (math.sqrt(sq) + 1) ** 2 if math.sqrt(sq) % 1 == 0 else -1


print(second(121))
